### DISCUSSION TITLE
Fail to render custom banner via http

### DISCUSSION THREAD

**Discussion by jeffscottward** [google/model-viewer#2107] [Q&A]
Hey all,

It was suggested to me to post this.

While trying to dev locally, I couldn’t get the HTML content for the custom banner on iOS-src to show up. It would just be a blank box.

The Apple custom banner docs suggest to use https to comply with security practices but isn’t clear if that means it won’t work outright without it.

I got it to work by quickly uploading a folder with the html file via the Piñata IPFS pinning service and then pointing to the {hash/folder/html-file} through the https cloudflare IPFS gateway.

see original code issue here
https://twitter.com/jeffscottward/status/1368642031610892294?s=21

Tldr;
Why can’t I just use http localhost to serve an HTML custom banner while in dev?

Follow up question:

(Using Next.js) If https is needed, how would I solve this problem so I can also properly test on my usb-c connected iPhone as I use Safari Dev Tools to test?

Typically on my phone I go to http://192.168.0.{X}:3000 to get hot-reload etc while deving.

**Comment by karanganesan** [comment]
> Why can’t I just use http localhost to serve an HTML custom banner while in dev?

TL;DR you can't sadly 😅
it's outside Apple's Quick Look Security Standards
so it's ignored

![20210313_090952](https://user-images.githubusercontent.com/6049925/111017921-2adb7000-83dc-11eb-96bf-192bfdc812cc.jpg)

Docs: https://developer.apple.com/documentation/arkit/adding_an_apple_pay_button_or_a_custom_action_in_ar_quick_look

> (Using Next.js) If https is needed, how would I solve this problem so I can also properly test on my usb-c connected iPhone as I use Safari Dev Tools to test?

Try this out 
"NextJS + HTTPS | For a Local Dev Server"
https://medium.com/responsetap-engineering/nextjs-https-for-a-local-dev-server-98bb441eabd7
